src/test_stack.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for stack.py."""
+
+
+def test_stack_init():
+    """Test for Stack init."""
+    from stack import Stack
+    new_stack = Stack()
+    assert new_stack.length == 0
```

Comment:
You guys need to fix your tests so there's only one assert statement per test function. The reason for this is so you can get a more granular understanding of what cases/scenarios fail and which don't.

A big problem is that if one assertion is false, the function exits and the rest of the tests are not evaluated, so you are denied more information about the success/failure of your tests. Separate those concerns. Be explicit.

And once again, use fixtures and parametrization. It can save you from so much of this messy multi-line test function malarkey.